src/test/java/com/klodnicki/Bike/v2/service/ChargingStationServiceHandlerTest.java

```diff
@@ -8,146 +8,143 @@
+        bikeServiceHandler = mock(BikeServiceHandler.class);
+        modelMapper = mock(ModelMapper.class);
+        chargingStationServiceHandler = new ChargingStationServiceHandler(chargingStationRepository, bikeServiceHandler,
+                modelMapper);
+        chargingStation = mock(ChargingStation.class);
+
+        when(chargingStationRepository.save(chargingStation)).thenReturn(chargingStation);
     }
 
     @Test
-    public void add_ShouldAddToDatabase_WhenGivenCorrectArguments() {
-        //given
-        Long expected = chargingStationRepository.count() + 1;
-        ChargingStation chargingStation = new ChargingStation();
-
-        //when
+    public void add_ShouldCallOnChargingStationRepositoryExactlyOnce_WhenChargingStationProvided() {
+        //Arrange - takes from @BeforeEach setUp()
+        //Act
         chargingStationServiceHandler.add(chargingStation);
-        Long actual = chargingStationRepository.count();
 
-        //then
-        assertEquals(expected, actual);
+        //Assert
+        verify(chargingStationRepository, times(1)).save(chargingStation);
     }
 
     @Test
     public void add_ShouldReturnStationForAdminResponseDTO_WhenGivenChargingStation() {
-        //given
-        ChargingStation chargingStation = new ChargingStation();
-        StationForAdminResponseDTO expected = modelMapper.map(chargingStation, StationForAdminResponseDTO.class);
+        //Arrange - takes from @BeforeEach setUp()
+        StationForAdminResponseDTO stationDTO = new StationForAdminResponseDTO();
+        when(modelMapper.map(chargingStation, StationForAdminResponseDTO.class)).thenReturn(stationDTO);
 
-        //when
+        //Act
         StationForAdminResponseDTO actual = chargingStationServiceHandler.add(chargingStation);
 
-        //then
-        assertNotNull(actual.getId());
-        expected.setId(actual.getId());
-        assertEquals(expected, actual);
+        //Assert
+        assertEquals(stationDTO, actual);
     }
 
     @Test
-    public void findAll_ShouldReturnListOfStationForAdminResponseDTO_WhenExistInDatabase() {
-        //given
+    public void findAll_ShouldReturnListOfStationForAdminResponseDTO_WhenBikesExistInDatabase() {
+        //Arrange - takes from @BeforeEach setUp()
+        List<ChargingStation> stations = new ArrayList<>();
         List<StationForAdminResponseDTO> expectedStationDTOS = new ArrayList<>();
+        StationForAdminResponseDTO stationDTO = new StationForAdminResponseDTO();
+        when(chargingStationRepository.findAll()).thenReturn(stations);
+
         for (int i = 0; i < 5; i++) {
-            ChargingStation chargingStation = new ChargingStation();
-            chargingStationRepository.save(chargingStation);
-            StationForAdminResponseDTO stationDTO = modelMapper.map(chargingStation, StationForAdminResponseDTO.class);
+            when(modelMapper.map(chargingStation, StationForAdminResponseDTO.class)).thenReturn(stationDTO);
```

Comment:
Why is that in the for loop? It definitely shouldn't be there.

Reply:
since it is about returning the List of DTOs I just used loop to create 5 instances and put them to the list.

Reply:
But why is the `when` method inside of the loop?

Reply:
My mistake, when method should be definetly outside the loop